### ISSUE TITLE
Question about Training Epochs Configuration and Best Practices

### ISSUE THREAD

**Issue by wangml71** [HCPLab-SYSU/DDP-WM#3]
Dear DDP-WM Authors,

First of all, thank you for your excellent work "DDP-WM: Disentangled Dynamics Prediction for Efficient World Models" and for open-sourcing the code. Your disentangled dynamics prediction paradigm is truly inspiring!

## Background

I am a Master's student at Southwest Jiaotong University, currently working on reproducing your DDP-WM approach for my research. I'm studying how your method achieves efficient world model learning through the three-stage training strategy (localization → primary_predictor → lrm).

## Questions

While going through the codebase and the paper, I noticed some discrepancies regarding the training epochs configuration, and I would greatly appreciate your guidance:

### 1. Training Epochs in Practice

- **Paper (Appendix C)**: mentions `100 training epochs`
- **Default config (`conf/train.yaml`)**: sets `epochs: 10`
- **Related work (DINO-WM)**: also uses `epochs: 100` in config, but publicly released checkpoints show much lower epoch numbers (e.g., `model_2.pth`)

**Questions:**
- How many epochs did you actually train for each stage in your published experiments?
- Is `epochs: 100` meant as an upper bound with early stopping, or did you train the full 100 epochs?
- For the three-stage training, what is the recommended `epochs_per_stage` configuration?

### 2. Computational Resources

- The SLURM config shows `gpu:A6000:2` (2× A6000 GPUs)
- In my preliminary tests on Push-T using my current setup (1× RTX5090, 22 CPUs, 110GB RAM), one epoch takes approximately 8 hours

**Questions:**
- Did you use more GPUs for the experiments reported in the paper?
- What was the actual training time for your published results?
- Are there any optimizations (e.g., mixed precision, gradient accumulation) that significantly speed up training?

### 3. Pre-trained Checkpoints

I noticed that DINO-WM provides publicly downloadable checkpoints, but I couldn't find pre-trained DDP-WM models.

**Questions:**
- Are you planning to release pre-trained checkpoints for the five environments (PointMaze, PushT, Wall, Rope, Granular)?
- If not, could you share the training curves or convergence behavior so we know when to stop training?

## Reproduction Setup

My current configuration:
- **Hardware**: 1× NVIDIA RTX5090 (32GB), 22 CPU cores, 110GB RAM
- **Dataset**: Push-T (pusht_noise)
- **Current config**: 
  ```yaml
  training:
    stages: ['localization', 'primary_predictor', 'lrm']
    epochs_per_stage: [10, 10, 10]
    batch_size: 32

### Acknowledgements

I sincerely appreciate your time in reading this email and addressing my questions. Your insights would be incredibly valuable to my research and will help ensure a faithful reproduction of your work.

Thank you again for your contribution to the community.

Best regards,

Mingling Wang
Master's Student
Southwest Jiaotong University

**Comment by ChunZheng2022** [comment]
@wangml71 Hi, I’ve also been trying to reproduce this work lately. I have a few suggestions regarding speeding up training that you might find helpful. Drawing from what’s outlined in the DINO-WM paper, you could try omitting the decoder during your experiments to boost training speed and cut down on VRAM usage. Additionally, the current dataloader contains some redundant logic. I ran into repeated out-of-memory (OOM) errors before, and after making simple tweaks to the dataloader, training speed improved noticeably as well.

**Comment by wangml71** [comment]
@ChunZheng2022 Hello, thank you for your reply. I’ve tried disabling decoder training, and I’ve also tried quantizing the model (such as fp16, bf16, etc.). I can see that the instantaneous training speed improves, but due to the data loading bottleneck, the training throughput still fluctuates in a sawtooth pattern, with the GPU stopping and waiting for data to load. As a result, the average training speed has not improved significantly. Could you talk in more detail about how you tuned the dataloader?

**Comment by ChunZheng2022** [comment]
@wangml71  
Hi! Here’s what I did:
In the original datasets/traj_dset.py, each sample only requires around 4 frames, but the code first decodes the entire video before slicing it up. I modified it to only load the frames needed for the current slice. This fixes some OOM errors and speeds up data loading.
So far I’ve only tested this on the PushT dataset and haven’t run into any issues. Feel free to let me know if you encounter any problems.

**Comment by wangml71** [comment]
> [@wangml71](https://github.com/wangml71) Hi! Here’s what I did: In the original datasets/traj_dset.py, each sample only requires around 4 frames, but the code first decodes the entire video before slicing it up. I modified it to only load the frames needed for the current slice. This fixes some OOM errors and speeds up data loading. So far I’ve only tested this on the PushT dataset and haven’t run into any issues. Feel free to let me know if you encounter any problems.

@ChunZheng2022 I changed the data loading module in Pusht to load only the frames that need to be trained at the moment. This indeed accelerated my training process (the average speed increased from 1.3 it/s to 5.0 it/s). Thank you for your suggestion. If you have any other good ideas, we can continue to discuss them.

**Comment by ChunZheng2022** [comment]
> > [@wangml71](https://github.com/wangml71) Hi! Here’s what I did: In the original datasets/traj_dset.py, each sample only requires around 4 frames, but the code first decodes the entire video before slicing it up. I modified it to only load the frames needed for the current slice. This fixes some OOM errors and speeds up data loading. So far I’ve only tested this on the PushT dataset and haven’t run into any issues. Feel free to let me know if you encounter any problems.
> 
> [@ChunZheng2022](https://github.com/ChunZheng2022) I changed the data loading module in Pusht to load only the frames that need to be trained at the moment. This indeed accelerated my training process (the average speed increased from 1.3 it/s to 5.0 it/s). Thank you for your suggestion. If you have any other good ideas, we can continue to discuss them.

Hi! Have you tried multi-GPU parallel training? I’ve found that multiple GPUs work better for this model than a single card with a large VRAM buffer. For example, training simultaneously with eight 22GB RTX 2080 Ti cards takes only around 40 minutes to complete one epoch on the PushT dataset.

**Comment by ChunZheng2022** [comment]
@wangml71 Hi! I recently tried transferring my work to a RTX Pro6000 device and noticed severe performance drops. This may be due to issues with the PyTorch and CUDA versions. I found that GPUs with the Blackwell architecture require CUDA 12.8 or newer. I recall your GPU is the NVIDIA RTX 5090, which is also based on Blackwell. Have you encountered similar issues?
